cmd/tendermint/commands/debug/dump.go

```diff
@@ -0,0 +1,128 @@
+package debug
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"time"
+
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+	cfg "github.com/tendermint/tendermint/config"
+	"github.com/tendermint/tendermint/libs/cli"
+	rpcclient "github.com/tendermint/tendermint/rpc/client"
+)
+
+var dumpCmd = &cobra.Command{
+	Use:   "dump [output-directory]",
+	Short: "Continuously poll a Tendermint process and dump debugging data into a single location",
+	Long: `Continuously poll a Tendermint process and dump debugging data into a single
+location at a specified frequency. At each frequency interval, an archived and compressed
+file will contain node debugging information including the goroutine and heap profiles
+if enabled.`,
+	Args: cobra.ExactArgs(1),
+	RunE: dumpCmdHandler,
+}
+
+func init() {
+	dumpCmd.Flags().UintVar(
+		&frequency,
+		flagFrequency,
+		30,
+		"The frequency (seconds) in which to poll, aggregate and dump Tendermint debug data",
+	)
+
+	dumpCmd.Flags().StringVar(
+		&profAddr,
+		flagProfAddr,
+		"",
+		"The profiling server address (<host>:<port>)",
+	)
+}
+
+func dumpCmdHandler(_ *cobra.Command, args []string) error {
+	outDir := args[0]
+	if outDir == "" {
+		return errors.New("invalid output directory")
+	}
+
+	if frequency == 0 {
+		return errors.New("frequency must be positive")
+	}
+
+	if _, err := os.Stat(outDir); os.IsNotExist(err) {
+		os.Mkdir(outDir, os.ModePerm)
+	}
+
+	rpc := rpcclient.NewHTTP(nodeRPCAddr, "/websocket")
+
+	home := viper.GetString(cli.HomeFlag)
+	conf := cfg.DefaultConfig()
+	conf = conf.SetRoot(home)
+	cfg.EnsureRoot(conf.RootDir)
+
+	dumpDebugData(outDir, conf, rpc)
+
+	ticker := time.NewTicker(time.Duration(frequency) * time.Second)
+	for range ticker.C {
```

Comment:
will I be able to stop the program by pressing `CTRL-C`?

Reply:
Yes :)